compute/compute.js

```diff
@@ -37,563 +37,36 @@ steal('can/util', 'can/util/bind', 'can/util/batch', function (can, bind) {
 	// 
 	// Each object-event pair is mapped so no duplicates will be listed.
 
-	// ### State
-	// 
-	// `can.__read` may call a function that calls `can.__read` again. For
-	// example, a compute can read another compute. To track each compute's
-	// `Observed` object (containing observable objects and events), we maintain
-	// a stack of Observed values for each call to `__read`.
-	var stack = [];
-
-	// ### can.__read
-	//
-	// With a given function and context, calls the function
-	// and returns the resulting value of the function as well
-	// as the observable properties and events that were read.
-	can.__read = function (func, self) {
+	can.compute = function (getterSetter, context, eventName, bindOnce) {
 
-		// Add an object that `can.__read` will write to.
-		stack.push({});
+		var internalCompute = new can.Compute(getterSetter, context, eventName, bindOnce);
 
-		var value = func.call(self);
+		var compute = function(val) {
+			if(arguments.length) {
+				return internalCompute.set(val);
+			}
 
-		// Example return value:
-		// `{value: 100, observed: Observed}`
-		return {
-			value: value,
-			observed: stack.pop()
+			return internalCompute.get();
 		};
-	};
 
-	// ### can.__reading
-	//
-	// When an observable value is read, it must call `can.__reading` to 
-	// broadcast which object and event should be listened to.
-	can.__reading = function (obj, event) {
-		// Add the observable object and the event
-		// that was read to the `Observed` object on
-		// the stack.
-		if (stack.length) {
-			stack[stack.length-1][obj._cid + '|' + event] = {
-				obj: obj,
-				event: event + ""
-			};
-		}
-
-	};
-
-	// ### can.__clearReading
-	//
-	// Clears and returns the current observables.
-	// This can be used to access a value without 
-	// it being handled as a regular `read`.
-	can.__clearReading = function () {
-		if (stack.length) {
-			var ret = stack[stack.length-1];
-			stack[stack.length-1] = {};
-			return ret;
-		}
-	};
-	// Specifies current observables.
-	can.__setReading = function (o) {
-		if (stack.length) {
-			stack[stack.length-1] = o;
-		}
-	};
-	can.__addReading = function(o){
-		if (stack.length) {
-			can.simpleExtend(stack[stack.length-1], o);
-		}
-	};
+		compute.bind = can.proxy(internalCompute.bind, internalCompute);
+		compute.unbind = can.proxy(internalCompute.unbind, internalCompute);
+		compute.isComputed = can.proxy(internalCompute.isComputed, internalCompute);
```

Comment:
In my playing with this branch I noticed that isComputed was no longer a `boolean` and was uber confused. Glad that's cleared up.